### PR TITLE
Setup config for chain connector plugin - Closes #7218

### LINKED ISSUE #7218: Setup configuration for chain connector plugin
### Description

Add all the private properties and set the ones that is configurable with the values coming from the config in the `init()`  and `load()` functions.

- Add configSchema for ChainConnectorPlugin as 
```ts
interface ChainConnectorPluginConfig {
	mainchainIPCPath: string;
	sidechainIPCPath: string;
	ccmBasedFrequency: number;
	livenessBasedFrequency: number;
}
```
- Add private `ccuCreationFrequency` as `{ ccm: number, liveness: number }` which comes from config
- Add private `_mainchainAPIClient` and `_sidechainAPIClient` (optional) to interact with mainchain/sidechain node
- Add private `sentCCUs` array to store sent CCU transaction objects
- Add private `lastCertifiedHeight` to be used in computation of CCU

### Acceptance Criteria

- Add unit test to test if all properties are properly set from config

### Additional Info

- Depends on #7217

### PR DESCRIPTION
**Pull request by ishantiw**
### What was the problem?

This PR resolves #7218

### How was it solved?

[🌱Setup configuration for chain connector plugin](https://github.com/LiskHQ/lisk-sdk/commit/0a0e5c0b81f3c21da60fd878cb02942011c5a33a)

### How was it tested?

Run `npm t` under `framework-plugins/lisk-framework-chain-connector-plugin`

### CODE REVIEW

File: framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts [hunk 1]
```diff
@@ -12,24 +12,59 @@
...
+import { configSchema } from './schemas';
+import { ChainConnectorPluginConfig } from './types';
 
-export class ChainConnectorPlugin extends BasePlugin {
+interface CCUFrequencyConfig {
+	ccm: number;
+	liveness: number;
+}
+
+type SentCCUs = Transaction[];
```
Comment (Incede) [inline framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts:25]: maybe move to `types.ts`

File: framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts [hunk 2]
```diff
@@ -12,24 +12,59 @@
...
+
+export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig> {
 	public name = 'chainConnector';
+	public configSchema = configSchema;
+	private _lastCertifiedHeight!: number;
+	private _ccuFrequency: CCUFrequencyConfig = {
+		ccm: CCM_BASED_CCU_FREQUENCY,
+		liveness: LIVENESS_BASED_CCU_FREQUENCY,
+	};
+	private _mainchainAPIClient!: apiClient.APIClient;
```
Comment (Incede) [inline framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts:35]: what's the reason for null assertion for `_mainchainAPIClient` ?

File: framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts [hunk 3]
```diff
@@ -12,24 +12,59 @@
...
+	ccm: number;
+	liveness: number;
+}
+
+type SentCCUs = Transaction[];
+
+export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig> {
 	public name = 'chainConnector';
+	public configSchema = configSchema;
+	private _lastCertifiedHeight!: number;
```
Comment (Incede) [inline framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts:30]: `Add private RPC rpc: { mainchain, sidechain? }`
Property mentioned in the issue but missing here.

File: framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts [hunk 4]
```diff
@@ -12,24 +12,59 @@
...
+	private _ccuFrequency: CCUFrequencyConfig = {
```
Comment (Incede) [inline framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts:31]: probably doesn't need to set deaults here if it is doing in `init`

**Review by Incede — CHANGES_REQUESTED** [review]

File: framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts (hunk 2 above)
Comment (ishantiw) [reply framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts:35]: its not assigned in constructor but in the load() so it throws lint error if I don't assign null assertion

**Review by has5aan — APPROVED** [review]

**Review by Incede — APPROVED** [review]

**Pull request merged**